molgenis-data/src/main/java/org/molgenis/data/decorator/DynamicRepositoryDecoratorRegistryImpl.java

```
@@ -0,0 +1,93 @@
+
+	public DynamicRepositoryDecoratorRegistryImpl(DataService dataService, RepositoryCollection repositoryCollection,
+			DecoratorConfigurationMetadata decoratorConfigurationMetadata)
+	{
+		this.dataService = requireNonNull(dataService);
+		this.repositoryCollection = requireNonNull(repositoryCollection);
+		this.decoratorConfigurationMetadata = requireNonNull(decoratorConfigurationMetadata);
+	}
+
+	@Override
+	public synchronized void addFactory(DynamicRepositoryDecoratorFactory factory)
+	{
+		String factoryId = factory.getId();
+		factories.put(factoryId, factory);
+	}
+
+	@Override
+	public Stream<String> getFactoryIds()
+	{
+		return factories.keySet().stream();
+	}
+
+	@Override
+	public synchronized Repository<Entity> decorate(Repository<Entity> repository)
+	{
+		String entityTypeId = repository.getEntityType().getId();
+
+		if (!entityTypeId.equals(DECORATOR_CONFIGURATION) && repositoryCollection.hasRepository(
+				decoratorConfigurationMetadata))
+		{
+			Query query = new QueryImpl().eq(ENTITY_TYPE_ID, entityTypeId);
+			DecoratorConfiguration configuration = dataService.findOne(DECORATOR_CONFIGURATION, query,
+					DecoratorConfiguration.class);
+
+			if (configuration != null)
+			{
+				repository = decorateRepository(repository, configuration);
+			}
+		}
+
+		return repository;
+	}
+
+	@SuppressWarnings("unchecked")
+	private Repository<Entity> decorateRepository(Repository<Entity> repository, DecoratorConfiguration configuration)
+	{
+		List<DynamicDecorator> decorators = configuration.getDecorators().collect(toList());
+		if (decorators.isEmpty())
+		{
+			return repository;
+		}
+
+		for (DynamicDecorator decorator : decorators)
+		{
+			DynamicRepositoryDecoratorFactory factory = factories.get(decorator.getId());
+			if (factory != null)
+			{
+				repository = factory.createDecoratedRepository(repository);
+			}
+			else
+			{
+				//FIXME: throw appropriate exception
```

Could you resolve this FIXME? In my opinion we should ignore this exception silently.